emop-maven-plugin/src/main/java/edu/cornell/AffectedSpecsMojo.java

```diff
@@ -224,26 +228,73 @@ protected String getClassPath() throws MojoExecutionException {
         return Writer.pathToString(getSureFireClassPath().getClassPath());
     }
 
-    private void makeSourcesFile(String sourceList, Set<String> impacted) {
+    /**
+     * Given a path to a class file, returns a path to its corresponding source file. Assumes a standard directory
+     * layout, i.e., one where the source for {@code com.abc.A} resides in {@code sourceDir/com/abc/A.java}.
+     * @param classFile the path to the class file
+     * @param classesDir the base class file directory
+     * @param sourceDir the base sources directory
+     * @return the path to the source file
+     */
+    private static Path classFileToSource(Path classFile, Path classesDir, Path sourceDir) {
+        Path parent = sourceDir.resolve(classesDir.relativize(classFile)).getParent();
+        return parent.resolve(classFile.getFileName().toString().replace(".class", ".java"));
+    }
+
+    private void makeSourcesFile(String sourceList, Set<String> impacted) throws MojoExecutionException {
         Set<String> classes = new HashSet<>();
+        Path testSourceDir = getTestSourceDirectory().toPath().toAbsolutePath();
+        Path sourceDir = Paths.get(getTestSourceDirectory().getAbsolutePath().replace(
+                "src" + File.separator + "test", "src" + File.separator + "main"));
+
         for (String klas : impacted) {
```

Comment:
Is it possible to call this something besides klas? Like impactedClass or something